crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -664,7 +793,36 @@ fn process_systems(
 /// Returns vector containing all pairs of indices of systems with ambiguous execution order,
 /// along with specific components that have triggered the warning.
 /// Systems must be topologically sorted beforehand.
-fn find_ambiguities(systems: &[impl SystemContainer]) -> Vec<(usize, usize, Vec<ComponentId>)> {
+fn find_ambiguities(
+    systems: &[impl SystemContainer],
+    crates_filter: &[String],
+) -> Vec<(usize, usize, Vec<ComponentId>)> {
+    fn should_ignore_ambiguity(
+        systems: &[impl SystemContainer],
+        index_a: usize,
+        index_b: usize,
+        crates_filter: &[String],
+    ) -> bool {
+        let system_a = &systems[index_a];
+        let system_b = &systems[index_b];
+
+        (match system_a.ambiguity_detection() {
+            AmbiguityDetection::Ignore => true,
+            AmbiguityDetection::Check => false,
+            AmbiguityDetection::IgnoreWithLabel(labels) => {
+                labels.iter().any(|l| system_b.labels().contains(l))
+            }
+        }) || (match system_b.ambiguity_detection() {
+            AmbiguityDetection::Ignore => true,
+            AmbiguityDetection::Check => false,
+            AmbiguityDetection::IgnoreWithLabel(labels) => {
+                labels.iter().any(|l| system_a.labels().contains(l))
+            }
+        }) || crates_filter
+            .iter()
+            .any(|s| system_a.name().starts_with(s) || system_b.name().starts_with(s))
```

Comment:
```suggestion
            .any(|s| system_a.name().starts_with(s) && system_b.name().starts_with(s))
```
I think Ambiguities should still be reported, if one of the two systems is not inside ingnored crates.

Reply:
Ambiguity is checked in pairs, so if I have `system_a` and `system_b` and I set `system_a` to be ignored, there is no point IMO in showing `system_b` as conflicting with `system_a` since `system_a` was ignored already. This can be seems as a bug by final users IMO.

Reply:
I have to disagree here. I see the `ignore_crates` field of `ReportExecutionOrderAmbiguities` as a way to hide the internal ambiguities of a crate(-combination).

Concrete Case:
1. A have a crate that doesn't match anything inside `ignore_crates`.
2. Inside the crate, I have a System that reads from `GlobalTransform` components in the PostUpdate Stage, and that depends that on having it set correctly.
3. This conflicts with the `transform_propagate_system` inside `bevy_transform`, which sets the `GlobalTransform` components to their correct Value.
4. I do not use a `.before(TransformSystem::TransformPropagate)` or `.after(TransformSystem::TransformPropagate)` or set the two systems as an intentional ambiguity.

This ambiguity will not be reported, because any ambiguities with `transform_propagate_system` are suppressed because it is part of `bevy_transform`.

This lulls me into the false impression, that my system will work correctly, and when I finally notice that I seem to have an ambiguity Bug, it's unnecessarily hard to find the actual conflict.

This all happened `ignore_crates` hides ambiguities **WITH** internal bevy systems, instead of **between** internal bevy systems.

One thing I must concede is that I suggested the wrong change.
This is the change I actually wanted:

```rs
       }) || (crates_filter.iter().any(|s| system_a.name().starts_with(s))
			 && crates_filter.iter().any(|s| system_a.name().starts_with(s)))
```

Reply:
> This all happened ignore_crates hides ambiguities WITH internal bevy systems, instead of between internal bevy systems.

I agree, the behaviour should be to ignore ambiguities within the crates on the blacklist, not ambiguities that may occur between other systems and the systems in those crates.

Reply:
You are right @MinerSebas, I got your point, so we should rework in a way that ignore crates really do ignore ambiguities between systems inside the given crate.

I'll submit a new commit changing this behavior.

Reply:
The following test code:
```rust
use bevy::{ecs::schedule::ReportExecutionOrderAmbiguities, prelude::*};

fn main() {
    env_logger::init();
    App::new()
        .add_plugins(DefaultPlugins)
        .insert_resource(ReportExecutionOrderAmbiguities::verbose())
        .add_system_to_stage(CoreStage::PostUpdate, ambiguous_system)
        .run();
}

fn ambiguous_system(mut q: ResMut<Assets<Texture>>) {}
```

Now produces the following output:
```
Sep 08 16:42:59.199  WARN bevy_ecs::schedule::stage: 4 unresolved ambiguities detected.
Sep 08 16:42:59.199  INFO bevy_ecs::schedule::stage: Execution order ambiguities detected, you might want to add an explicit dependency relation between some of these systems:
1. Parallel systems:
1.0 - Systems:
     ["bevy_sprite::sprite::sprite_system", "ambiguity_test::ambiguous_system", "bevy_ui::widget::text::text_system", "bevy_ui::widget::image::image_node_system", "bevy_text::text2d::text2d_system"]
   - Conflicts on the following components/resources:
     ["bevy_asset::assets::Assets<bevy_render::texture::texture::Texture>"]
```

Since `ambiguous_system` conflicts with bevy internal crates, even if the bevy crates are ignored by default, we just ignore conflicts between bevy crates, so this way users can know when their systems are conflicting with bevy internal systems.